SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/ManPages/tsManPagesLibGUI/tsWxMouseState.man

Help on module tsWxMouseState:

NAME
    tsWxMouseState

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxMouseState.py

DESCRIPTION
    tsWxMouseState.py - Class to hold information about mouse button and
    modifier key states and is what is returned from wx.GetMouseState.

CLASSES
    tsWxKeyboardState.KeyboardState(__builtin__.object)
        MouseState
    
    class MouseState(tsWxKeyboardState.KeyboardState)
     |  This class s used to hold information about mouse button and modifier
     |  key states and is what is returned from wx.GetMouseState.
     |  
     |  Method resolution order:
     |      MouseState
     |      tsWxKeyboardState.KeyboardState
     |      __builtin__.object
     |  
     |  Methods defined here:
     |  
     |  Aux1IsDown(self)
     |      Returns true if the first extra mouse button is currently down.
     |  
     |  Aux2IsDown(self)
     |      Returns true if the second extra mouse button is currently down.
     |  
     |  GetX(self)
     |      Returns X coordinate of the physical mouse event position.
     |  
     |  GetY(self)
     |      Returns Y coordinate of the physical mouse event position.
     |  
     |  LeftIsDown(self)
     |      Returns true if the left mouse button is currently down.
     |  
     |  MiddleIsDown(self)
     |      Returns true if the middle mouse button is currently down.
     |  
     |  RightIsDown(self)
     |      Returns true if the right mouse button is currently down.
     |  
     |  SetAux1IsDown(self, down)
     |      Sets the first extra mouse button state at the time of the event.
     |  
     |  SetAux2IsDown(self, down)
     |      Sets the second extra mouse button state at the time of the event.
     |  
     |  SetLeftIsDown(self, down)
     |      Sets the Left mouse button state at the time of the event.
     |  
     |  SetMiddleIsDown(self, down)
     |      Sets the Middle mouse button state at the time of the event.
     |  
     |  SetRightIsDown(self, down)
     |      Sets the Right mouse button state at the time of the event.
     |  
     |  SetX(self, x)
     |      Sets the X coordinate of the physical mouse event position.
     |  
     |  SetY(self, y)
     |      Sets the Y coordinate of the physical mouse event position.
     |  
     |  __del__(self)
     |  
     |  __init__(self)
     |      Constructs a wx.MouseState.
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxKeyboardState.KeyboardState:
     |  
     |  AltDown(self)
     |      Returns true if the Alt key is pressed.
     |      
     |      Notice that GetModifiers() should usually be used instead of this one.
     |  
     |  CmdDown(self)
     |      Returns true if the key used for command accelerators is pressed.
     |      
     |      Cmd is a pseudo key which is Control for PC and Unix platforms but
     |      Apple (or Command) key under Macs: it makes often sense to use it
     |      instead of ControlDown() because Command key is used for the same
     |      thing under Mac as Control elsewhere (even though Control still
     |      exists, it is usually not used for the same purpose under Mac).
     |      
     |      Notice that GetModifiers() should usually be used instead of this
     |      one. However, the Apple "Cmd" (Command) is Not supported, by
     |      Ncurses, on computer platforms running Mac OS X, Linux or Windows.
     |  
     |  ControlDown(self)
     |      Returns true if the Control key is pressed.
     |      
     |      This function does not distinguish between right and left control
     |      keys.
     |      
     |      In portable code you usually want to use CmdDown() to automatically
     |      test for the more frequently used Command key (and not the rarely
     |      used Control one) under Mac.
     |      
     |      Notice that GetModifiers() should usually be used instead of this one.
     |  
     |  GetModifiers(self)
     |      Return the bit mask of all pressed modifier keys.
     |      
     |      The return value is a combination of wxMOD_ALT, wxMOD_CONTROL,
     |      wxMOD_SHIFT and wxMOD_META bit masks. Additionally, wxMOD_NONE
     |      is defined as 0, i.e. corresponds to no modifiers
     |      (see HasModifiers()) and wxMOD_CMD is either wxMOD_CONTROL
     |      (MSW and Unix) or wxMOD_META (Mac), see CmdDown().
     |      See wxKeyModifier for the full list of modifiers.
     |      
     |      Notice that this function is easier to use correctly than, for
     |      example, ControlDown() because when using the latter you also
     |      have to remember to test that none of the other modifiers is pressed:
     |      
     |      if ( ControlDown() && !AltDown() && !ShiftDown() && !MetaDown() )
     |          ... handle Ctrl-XXX ...
     |      
     |      and forgetting to do it can result in serious program bugs (e.g.
     |      program not working with European keyboard layout where AltGr key
     |      which is seen by the program as combination of CTRL and ALT is used).
     |      On the other hand, you can simply write:
     |      
     |      if ( GetModifiers() == wxMOD_CONTROL )
     |          ... handle Ctrl-XXX ...
     |      
     |      with this function.
     |  
     |  HasModifiers(self)
     |      Returns true if any modifiers at all are pressed.
     |      
     |      This is equivalent to GetModifiers() != wxMOD_NONE.
     |  
     |  MetaDown(self)
     |      Returns true if the Meta/Windows/Apple key is pressed.
     |      
     |      This function tests the state of the key traditionally called Meta
     |      under Unix systems, Windows keys under MSW and Apple, or Command,
     |      key under Mac.
     |      
     |      Notice that GetModifiers() should usually be used instead of this one.
     |      
     |      See also:
     |      
     |      CmdDown()
     |      
     |      Reimplemented in wxMouseEvent.
     |      
     |      Notice: Apple "Cmd" (Command) Not supported, by Ncurses, on computer
     |      platforms running Mac OS X, Linux or Windows.
     |  
     |  SetAltDown(self, down)
     |      Sets the Alt key down state at the time of the event.
     |  
     |  SetCmdDown(self, down)
     |      Sets the Cmd key down state at the time of the event.
     |      "Cmd" is a pseudo key which is the same as Control for PC and Unix
     |      platforms but the special "Apple" (a.k.a as "Command") key on Macs.
     |      
     |      Notice: Apple "Cmd" (Command) Not supported, by Ncurses, on computer
     |      platforms running Mac OS X, Linux or Windows.
     |  
     |  SetControlDown(self, down)
     |      Sets the Control key down state at the time of the event.
     |  
     |  SetMetaDown(self, down)
     |      Sets the Meta key down state at the time of the event.
     |      
     |      Notice: Apple "Cmd" (Command) Not supported, by Ncurses, on computer
     |      platforms running Mac OS X, Linux or Windows.
     |  
     |  SetShiftDown(self, down)
     |      Sets the Shift key down state at the time of the event.
     |  
     |  ShiftDown(self)
     |      Returns true if the Shift key is pressed.
     |      
     |      This function does not distinguish between right and left shift keys.
     |      
     |      Notice that GetModifiers() should usually be used instead of this one.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from tsWxKeyboardState.KeyboardState:
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/01/2013'
    __header__ = '\n\ntsWxMouseState, v1.0.0 (build 04/01/2013)\n\n  A... ...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxMouseState, v1.0.0 (build 04/01/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxMouseState'
    __version__ = '1.0.0'
    mainTitleVersionDate = 'tsWxMouseState, v1.0.0 (build 04/01/2013)'

VERSION
    1.0.0

DATE
    04/01/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsDemoArchive/tsTutorialDeveloperSandbox2x/tsLibGUI/tsWxPkg/src/tsWxMouseState.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  7:33:10 AM rsg>"
'''
tsWxMouseState.py - Class to hold information about mouse button and
modifier key states and is what is returned from wx.GetMouseState.
'''
#################################################################
#
# File: tsWxMouseState.py
#
# Purpose:
#
#    Class to hold information about mouse button and modifier key
#    states and is what is returned from wx.GetMouseState.
#
# Usage (example):
#
#    # Import
#
#    from tsWxMouseState import MouseState
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    2010/09/18 rsg Converted the class instance variables to
#                   class variables so that the MouseState is
#                   global across all referencing modules.
#
#    2010/11/06 rsg Reverted to class instance variables.
#
#    2010/11/08 rsg Removed properties to conform with wxWidgets API.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxMouseState'
__version__   = '1.0.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

##import tsWxGlobals as wx # Invalid import because of circular dependency.
import tsCxGlobals as cx

#---------------------------------------------------------------------------

##DEBUG = True # Avoids tsWx import because of circular dependency.
DEBUG = cx.ThemeToUse['Troubleshooting']['Debug_GUI_Launch'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Progress'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Termination'] | \
        cx.ThemeToUse['Troubleshooting']['Debug_GUI_Exceptions']

##VERBOSE = True # Avoids tsWx import because of circular dependency.
VERBOSE = cx.ThemeToUse['Troubleshooting']['Debug_GUI_Configuration']

from tsWxKeyboardState import KeyboardState

#---------------------------------------------------------------------------

class MouseState(KeyboardState):
    '''
    This class s used to hold information about mouse button and modifier
    key states and is what is returned from wx.GetMouseState.
    '''
    # Class variables

    #-----------------------------------------------------------------------

    def __init__(self):
        '''
        Constructs a wx.MouseState.
        '''
##        theClass = 'MouseState'
##        # Create class membership flag self.thisown
##        self.tsRegisterClassNameAndMembershipFlag(theClass)

        KeyboardState.__init__(self)

        self.ts_LeftIsDown = False
        self.ts_MiddleIsDown = False
        self.ts_RightIsDown = False
        self.ts_Aux1IsDown = False
        self.ts_Aux2IsDown = False
        self.ts_X = 0
        self.ts_Y = 0

    #-----------------------------------------------------------------------

    def __del__(self):
        '''
        '''
        del self

    #-----------------------------------------------------------------------
 
    def GetX(self):
        '''
        Returns X coordinate of the physical mouse event position.
        '''
        return (self.ts_X)

    #-----------------------------------------------------------------------

    def GetY(self):
        '''
        Returns Y coordinate of the physical mouse event position.
        '''
        return (self.ts_Y)

    #-----------------------------------------------------------------------
 
    def LeftIsDown(self):
        '''
        Returns true if the left mouse button is currently down.
        '''
        return (self.ts_LeftIsDown)

    #-----------------------------------------------------------------------
 
    def MiddleIsDown(self):
        '''
        Returns true if the middle mouse button is currently down.
        '''
        return (self.ts_MiddleIsDown)

    #-----------------------------------------------------------------------
 
    def RightIsDown(self):
        '''
        Returns true if the right mouse button is currently down.
        '''
        return (self.ts_RightIsDown)

    #-----------------------------------------------------------------------
 
    def Aux1IsDown(self):
        '''
        Returns true if the first extra mouse button is currently down.
        '''
        return (self.ts_Aux1IsDown)

    #-----------------------------------------------------------------------
 
    def Aux2IsDown(self):
        '''
        Returns true if the second extra mouse button is currently down.
        '''
        return (self.ts_Aux2IsDown)

    #-----------------------------------------------------------------------
 
    def SetLeftIsDown(self, down):
        '''
        Sets the Left mouse button state at the time of the event.
        '''
        self.ts_LeftIsDown = down

    #-----------------------------------------------------------------------
 
    def SetMiddleIsDown(self, down):
        '''
        Sets the Middle mouse button state at the time of the event.
        '''
        self.ts_MiddleIsDown = down

    #-----------------------------------------------------------------------
 
    def SetRightIsDown(self, down):
        '''
        Sets the Right mouse button state at the time of the event.
        '''
        self.ts_RightIsDown = down

    #-----------------------------------------------------------------------
 
    def SetAux1IsDown(self, down):
        '''
        Sets the first extra mouse button state at the time of the event.
        '''
        self.ts_Aux1IsDown = down

    #-----------------------------------------------------------------------
 
    def SetAux2IsDown(self, down):
        '''
        Sets the second extra mouse button state at the time of the event.
        '''
        self.ts_Aux2IsDown = down

    #-----------------------------------------------------------------------
 
    def SetX(self, x):
        '''
        Sets the X coordinate of the physical mouse event position.
        '''
        self.ts_X = x

    #-----------------------------------------------------------------------

    def SetY(self, y):
        '''
        Sets the Y coordinate of the physical mouse event position.
        '''
        self.ts_Y = y

#---------------------------------------------------------------------------
 
if __name__ == '__main__':

    print(__header__)

    myState = MouseState()

    print('***** Initialized Modifier Key State ***')
    print('AltDown = %s' % myState.AltDown())
    print('CmdDown = %s' % myState.CmdDown())
    print('ControlDown = %s' % myState.ControlDown())
    print('MetaDown = %s' % myState.MetaDown())
    print('ShiftDown = %s' % myState.ShiftDown())

    print('***** Initialized Mouse Button State ***')
    print('LeftIsDown = %s' % myState.LeftIsDown)
    print('MiddleIsDown = %s' % myState.MiddleIsDown)
    print('RightIsDown = %s' % myState.RightIsDown)
    print('Aux1IsDown = %s' % myState.Aux1IsDown)
    print('Aux2IsDown = %s' % myState.Aux2IsDown)

    print('***** Initialized Mouse Position State ***')
    print('GetX = %d' % myState.GetX())
    print('GetY = %d' % myState.GetY())

    downStates = [True, False]
    for state in downStates:
        print('\n')

        print('***** Modifier Key States Set %s ***' % state)
        myState.SetAltDown(state)
        myState.SetCmdDown(state)
        myState.SetControlDown(state)
        myState.SetMetaDown(state)
        myState.SetShiftDown(state)

        print('***** Mouse Button States Set %s ***' % state)
        myState.SetLeftIsDown(state)
        myState.SetMiddleIsDown(state)
        myState.SetRightIsDown(state)
        myState.SetAux1IsDown(state)
        myState.SetAux2IsDown(state)

        print('***** Mouse Position States Set %s ***' % state)
        myState.SetX(myState.GetX() + 1234)
        myState.SetY(myState.GetY() + 5678)

        print('***** Updated Modifier Key States Set %s ***' % state)
        print('AltDown = %s' % myState.AltDown())
        print('CmdDown = %s' % myState.CmdDown())
        print('ControlDown = %s' % myState.ControlDown())
        print('MetaDown = %s' % myState.MetaDown())
        print('ShiftDown = %s' % myState.ShiftDown())

        print('***** Updated Mouse Button States Set %s ***' % state)
        print('LeftIsDown = %s' % myState.LeftIsDown())
        print('MiddleIsDown = %s' % myState.MiddleIsDown())
        print('RightIsDown = %s' % myState.RightIsDown())
        print('Aux1IsDown = %s' % myState.Aux1IsDown())
        print('Aux2IsDown = %s' % myState.Aux2IsDown())

        print('***** Updated Mouse Position States Set %s ***' % state)
        print('GetX = %d' % myState.GetX())
        print('GetY = %d' % myState.GetY())
